langchain/src/prompts/few_shot.ts

```diff
@@ -227,3 +234,218 @@ export class FewShotPromptTemplate
     });
   }
 }
+
+export interface FewShotChatMessagePromptTemplateInput
+  extends BasePromptTemplateInput<InputValues> {
+  /**
+   * Examples to format into the prompt. Exactly one of this or
+   * {@link exampleSelector} must be
+   * provided.
+   */
+  examples?: Example[];
+
+  /**
+   * An {@link BaseMessagePromptTemplate} | {@link BaseChatPromptTemplate} used to format a single example.
+   */
+  examplePrompt: BaseMessagePromptTemplate | BaseChatPromptTemplate;
+
+  /**
+   * String separator used to join the prefix, the examples, and suffix.
+   *
+   * @defaultValue `"\n\n"`
+   */
+  exampleSeparator?: string;
+
+  /**
+   * An {@link BaseExampleSelector} Examples to format into the prompt. Exactly one of this or
+   * {@link examples} must be
+   * provided.
+   */
+  exampleSelector?: BaseExampleSelector | undefined;
+
+  /**
+   * A prompt template string to put before the examples.
+   *
+   * @defaultValue `""`
+   */
+  prefix?: string;
+
+  /**
+   * A prompt template string to put after the examples.
+   *
+   * @defaultValue `""`
+   */
+  suffix?: string;
+
+  /**
+   * The format of the prompt template. Options are: 'f-string'
+   *
+   * @defaultValue `f-string`
+   */
+  templateFormat?: TemplateFormat;
+
+  /**
+   * Whether or not to try validating the template on initialization.
+   *
+   * @defaultValue `true`
+   */
+  validateTemplate?: boolean;
+}
+
+/**
+ * Chat prompt template that contains few-shot examples.
+ * @augments BasePromptTemplateInput
+ * @augments FewShotChatMessagePromptTemplateInput
+ */
+export class FewShotChatMessagePromptTemplate<
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    RunInput extends InputValues = any,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    PartialVariableName extends string = any
+  >
+  extends BaseChatPromptTemplate
+  implements FewShotChatMessagePromptTemplateInput
+{
+  lc_serializable = true;
+
+  examples?: InputValues[];
+
+  exampleSelector?: BaseExampleSelector | undefined;
+
+  examplePrompt: BaseMessagePromptTemplate | BaseChatPromptTemplate;
+
+  suffix = "";
+
+  exampleSeparator = "\n\n";
+
+  prefix = "";
+
+  templateFormat: TemplateFormat = "f-string";
+
+  validateTemplate = true;
+
+  _getPromptType(): "few_shot_chat" {
+    return "few_shot_chat";
+  }
+
+  constructor(fields: FewShotChatMessagePromptTemplateInput) {
+    super(fields);
+    Object.assign(this, fields);
```

Comment:
Do this explicitly, otherwise people can do dangerous things like override instance methods